### ISSUE TITLE
Support XDG Base Directory Specifications

### ISSUE THREAD

**Issue by GustavoMeza** [xalanq/cf-tool#92]
_XDG Base Directory Specification_ enables you to have a cleaner home directory by grouping config files for all programs under one common directory. This is useful for everyone that wants to keep a clean home directory.

[Full specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)

**Comment by infalmo** [comment]
Great idea! It would be much better if this was followed

**Comment by SubeetKD** [comment]
you can do it by editing the cf.go file in top area there is config location. Just edit it.

**Comment by useredsa** [comment]
> you can do it by editing the cf.go file in top area there is config location. Just edit it.

 I believe  it's a little bit more difficult. Instead of hardcoding a path, you need to look for the environment variable and use a default value if it's not present.

**Comment by akshettrj** [comment]
You can use my fork, it's XDG Base compliant.

**Comment by useredsa** [comment]
> You can use my fork, it's XDG Base compliant.

Why don't you pull request?

**Comment by akshettrj** [comment]
> > You can use my fork, it's XDG Base compliant.
> 
> Why don't you pull request?

I am not sure if the code will work for Windows and I think they are also making this for windows. Thats why I did not PR. I have no idea how Windows work.

**Comment by infalmo** [comment]
> > You can use my fork, it's XDG Base compliant.
> 
> Why don't you pull request?

I don't think this project is being maintained. In the meantime, you can give my spinoff a try: github.com/cp-tools/cpt

**Comment by akshettrj** [comment]
> > > You can use my fork, it's XDG Base compliant.
> > 
> > 
> > Why don't you pull request?
> 
> I don't think this project is being maintained. In the meantime, you can give my spinoff a try: github.com/cp-tools/cpt

I don't know if it is a problem only for me that's why I am posting it here. I tried to look at all of the configs and options but the problem with both the tools (xalanq's and yours) is that in whichever directory (let's say /A) I try to run the command, both the tools form subdirectories within /A. Instead, I want to set a root directory and want the tool to download the test cases in all within that directory irrespective of where I am running the command from. I did try to set the root folder in xalanq's tool (by giving it an absolute path) but it did not work the way I expected. Can you please guide me

**Comment by infalmo** [comment]
My tool allows you to set it, but you'll have to manually edit the configuration file (will fix this).
Open the `codeforces.yaml` file present in `cp-tools/cpt` in the config folder and add the following lines:
```
fetch:
  problemFolderPath:
  - '/home/username/Documents'
  - 'codeforces/{{ .Arg.Class }}/{{ .Arg.Group }}/{{ .Arg.Contest }}/{{ .Arg.Problem }}'
```

ofc, you can change it to how you deem fit (the one I've done is for linux). Basically, all the provided strings are merged into one and the problem folder is created in that directory. The default configuration don't specify the base folder (the first line in the parameters).

Hope your query has been resolved.

**Comment by infalmo** [comment]
Also, raising the issue in my repo would be good, if the issue is not resolved

**Comment by akshettrj** [comment]
> My tool allows you to set it, but you'll have to manually edit the configuration file (will fix this).
> Open the `codeforces.yaml` file present in `cp-tools/cpt` in the config folder and add the following lines:
> 
> ```
> fetch:
>   problemFolderPath:
>   - '/home/username/Documents'
>   - 'codeforces/{{ .Arg.Class }}/{{ .Arg.Group }}/{{ .Arg.Contest }}/{{ .Arg.Problem }}'
> ```
> 
> ofc, you can change it to how you deem fit (the one I've done is for linux). Basically, all the provided strings are merged into one and the problem folder is created in that directory. The default configuration don't specify the base folder (the first line in the parameters).
> 
> Hope your query has been resolved.

Oh! Thank you so much. Its really a good tool that you have. 👌😊
